File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RReachableQuestList.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RReachableQuestList(
        int total,
        List<Quest> quests
) {
    public record Quest(
            String id,
            String title,
            String subtitle,
            String chapterId,
            String chapterTitle,
            String groupId,
            String groupTitle,
            double x,
            double y,
            boolean visible,
            boolean started,
            boolean completed,
            boolean startable,
            int progress,
            int taskCount,
            int rewardCount,
            int dependencyCount,
            List<String> dependencyIds
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/model/RReachableQuestList.kt
package calebxzhou.rdi.mc.common2.mcp.model

@JvmRecord
data class RReachableQuestList(
    val total: Int,
    val quests: List<Quest>
) {
    @JvmRecord
    data class Quest(
        val id: String,
        val title: String,
        val subtitle: String,
        val chapterId: String,
        val chapterTitle: String,
        val groupId: String,
        val groupTitle: String,
        val x: Double,
        val y: Double,
        val visible: Boolean,
        val started: Boolean,
        val completed: Boolean,
        val startable: Boolean,
        val progress: Int,
        val taskCount: Int,
        val rewardCount: Int,
        val dependencyCount: Int,
        val dependencyIds: List<String>
    )
}
